components/ai_chat/core/browser/engine/conversation_api_v2_client_unittest.cc

```diff
@@ -0,0 +1,944 @@
+
+        // Verify messages content matches expected
+        const base::Value::List* messages_list = body_dict.FindList("messages");
+        EXPECT_TRUE(messages_list);
+        EXPECT_EQ(messages_list->size(), 1u);
+        const base::Value::Dict* message_dict = (*messages_list)[0].GetIfDict();
+        EXPECT_TRUE(message_dict);
+        EXPECT_EQ(*message_dict, expected_content);
+
+        // Simulate streaming chunk
+        auto chunk_dict = base::test::ParseJsonDict(R"({
+          "model": "chat-claude-sonnet",
+          "choices": [{
+            "delta": {"content": "chunk text"}
+          }]
+        })");
+        data_received_callback.Run(
+            base::ok(base::Value(std::move(chunk_dict))));
+
+        // Simulate completion
+        auto completion_dict = base::test::ParseJsonDict(R"({
+          "model": "chat-claude-sonnet",
+          "choices": [{
+            "message": {"content": "premium response"}
+          }]
+        })");
+        std::move(result_callback)
+            .Run(api_request_helper::APIRequestResult(
+                200, base::Value(std::move(completion_dict)), {}, net::OK,
+                GURL()));
+
+        run_loop.Quit();
+        return Ticket();
+      });
+
+  // Callbacks should be passed through and translated from APIRequestHelper
+  // format.
+  Sequence seq;
+  EXPECT_CALL(mock_callbacks, OnDataReceived(_))
+      .InSequence(seq)
+      .WillOnce([&](EngineConsumer::GenerationResultData result) {
+        ASSERT_TRUE(result.event);
+        EXPECT_TRUE(result.event->is_completion_event());
+        EXPECT_EQ(result.event->get_completion_event()->completion,
+                  "chunk text");
+      });
+  EXPECT_CALL(mock_callbacks, OnCompleted(_))
+      .WillOnce([&](EngineConsumer::GenerationResult result) {
+        ASSERT_TRUE(result.has_value());
+        EXPECT_TRUE(result->event);
+        EXPECT_TRUE(result->event->is_completion_event());
+        EXPECT_EQ(result->event->get_completion_event()->completion,
+                  expected_completion_response);
+      });
+
+  // Begin request
+  client_->PerformRequest(
+      std::move(messages), "" /* selected_language */, std::nullopt,
+      /* oai_tool_definitions */ std::nullopt, /* preferred_tool_name */
+      mojom::ConversationCapability::CHAT,
+      base::BindRepeating(&MockCallbacks::OnDataReceived,
+                          base::Unretained(&mock_callbacks)),
+      base::BindOnce(&MockCallbacks::OnCompleted,
+                     base::Unretained(&mock_callbacks)));
+
+  run_loop.Run();
+  testing::Mock::VerifyAndClearExpectations(client_.get());
+  testing::Mock::VerifyAndClearExpectations(mock_request_helper);
+  testing::Mock::VerifyAndClearExpectations(credential_manager_.get());
+}
+
+TEST_F(ConversationAPIV2ClientUnitTest, PerformRequest_NonPremium) {
+  // Performs the same test as Premium, verifying that nothing else changes
+  // apart from request headers (and request url).
```

Comment:
Can we parameterize the premium/non_premium and also `brave_capability` that sneaked in this case.

Reply:
Exactly pattern from v1 so I'll keep it as is, as refactoring is out of scope.